packages/@aws-cdk/core/lib/deps.ts

```diff
@@ -13,18 +19,55 @@ type Element = CfnResource | Stack;
  * - Try to find the deepest common stack between the two elements
  * - If there isn't a common stack, it means the elements belong to two
  *   disjoined stack-trees and therefore we apply the dependency at the
- *   assembly/app level between the two topl-level stacks.
+ *   assembly/app level between the two top-level stacks.
  * - If we did find a common stack, we apply the dependency as a CloudFormation
  *   "DependsOn" between the resources that "represent" our source and target
  *   either directly or through the AWS::CloudFormation::Stack resources that
  *   "lead" to them.
  *
- * @param source The source resource/stack (the depedent)
+ * @param source The source resource/stack (the dependent)
  * @param target The target resource/stack (the dependency)
  * @param reason Optional resource to associate with the dependency for
  * diagnostics
  */
-export function addDependency<T extends Element>(source: T, target: T, reason?: string) {
+export function addDependency(source: IElement, target: IElement, reason: string='dependency added using stack.addDependency()') {
+  operateOnDependency(DependencyOperation.ADD, source, target, reason);
+}
+
+/**
+ * Removes a dependency between two resources or stacks, across stack and nested
+ * stack boundaries.
+ *
+ * The algorithm consists of:
+ * - Try to find the deepest common stack between the two elements
+ * - If there isn't a common stack, it means the elements belong to two
+ *   disjoined stack-trees and therefore we applied the dependency at the
+ *   assembly/app level between the two top-level stacks; remove it there.
+ * - If we did find a common stack, we applied the dependency as a CloudFormation
+ *   "DependsOn" between the resources that "represent" our source and target
+ *   either directly or through the AWS::CloudFormation::Stack resources that
+ *   "lead" to them and must remove it there.
+ *
+ * @param source The source resource/stack (the dependent)
+ * @param target The target resource/stack (the dependency)
+ * @param reason Optional description to associate with the dependency for
+ * diagnostics
+ */
+export function removeDependency(source: IElement, target: IElement, reason: string='dependency added using stack.removeDependency()') {
```

Comment:
why the default value for `reason`? It doesn't look like we need this here.

Reply:
The default value of `'dependency added using stack.removeDependency()'` is confusing me. I assume this is a typo, but this reinforces my point that I don't think this default value is necessary.

Reply:
yeah, that's a mistake. Will fix.